src/app/file-browser/components/file-list/file-list.component.ts

```diff
@@ -200,6 +207,26 @@ export class FileListComponent
 		}
 	}
 
+	async ngOnChanges() {
+		this.currentFolder =
+			this.dataFolder || this.route.snapshot.data.currentFolder;
+		if (this.dataFolder) {
```

Comment:
The variable names here don't make it super easy to understand what is going on; this isn't your fault, but it does mean that we unfortunately might need to rely on comments to help explain what's happening and why.

(If we had infinite time I'd suggest we potentially come up with better names -- what is dataFolder exactly?  Does `currentFolder` mean `renderedFolder`?)

Anyway, let's at least toss in a few comments for future usses